### PR TITLE
CASMPET-5501: Match on BOS Operator pods

### PR DESCRIPTION
**Pull request by jsollom-hpe**
## Summary and Scope

Loosen the matching criteria, so that it matches on the BOS operator
pods, too, not just the BOS API server pods. This improves the security
coverage by extending the security context mutations to additional BOS
pods.

## Issues and Related PRs

_List and characterize relationship to Jira/Github issues and other pull requests. Be sure to list dependencies._

* Resolves CASMPET-5501

## Testing

_List the environments in which these changes were tested._

### Tested on:

  * Mug

### Test description:

I made this policy change. I then deleted the BOS V2 pods and some them restart with the security context added. Then, I tested that the BOS V2 micro-service functioned correctly while under these new security mutations. I ran through the entire BOS V2 happy path testing.

- Were the install/upgrade-based validation checks/tests run (goss tests/install-validation doc)? Nope.
- Were continuous integration tests run? If not, why? No, because we don't have any.
- Was upgrade tested? No, it was a policy change.
- Was downgrade tested? No, it was a policy change.
- Were new tests (or test issues/Jiras) created for this change? No.

## Risks and Mitigations

Low. We're expanding our security coverage surface.

## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [x] Copyrights updated
- [x] License file intact
- [x] Target branch correct
- [x] CHANGELOG.md updated
- [x] Testing is appropriate and complete, if applicable
- [x] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

File: charts/kyverno-policies/templates/pod-sec-ctxt-cray-bos.yaml [hunk 1]
```diff
@@ -24,7 +24,6 @@ spec:
         selector:
           matchLabels:
             app.kubernetes.io/instance: cray-bos
-            app.kubernetes.io/name: cray-bos
```
Comment (jeremy-duckworth) [inline charts/kyverno-policies/templates/pod-sec-ctxt-cray-bos.yaml:27]: A matching validation policy exists that needs to mirror this change. See https://github.com/Cray-HPE/cray-kyverno-policies/blob/CASMPET-5501/charts/kyverno-policies/templates/validation/validate-pod-sec-ctxt-cray-bos.yaml.

**Review by jeremy-duckworth — CHANGES_REQUESTED** [review]

File: charts/kyverno-policies/templates/pod-sec-ctxt-cray-bos.yaml (hunk 1 above)
Comment (jsollom-hpe) [reply charts/kyverno-policies/templates/pod-sec-ctxt-cray-bos.yaml:27]: I modified the validation policy, too. Thanks.

**Comment by jsollom-hpe** [comment]
@sriram-hpe @jeremy-duckworth , do I need to bump the version? The version checker is giving me an error.

**Comment by sriram-hpe** [comment]
> @sriram-hpe @jeremy-duckworth , do I need to bump the version? The version checker is giving me an error.

Yes. please bump the version to 1.2.0 to align with kyverno chart(current version of kyverno is 1.2.0)

**Review by jeremy-duckworth — APPROVED** [review]

**Pull request merged**